sound/soc/soc-topology.c

```diff
@@ -2135,11 +2049,8 @@ static int soc_tplg_pcm_elems_load(struct soc_tplg *tplg,
 
 		/* create the FE DAIs and DAI links */
 		ret = soc_tplg_pcm_create(tplg, _pcm);
-		if (ret < 0) {
-			if (!abi_match)
-				devm_kfree(tplg->dev, _pcm);
+		if (ret < 0)
```

Comment:
this looks wrong. `abi_match == false` when `pcm_new_ver()` above was called. In the original logic it was allocating a temporary `pcm` object which was unconditionally freed within this very block of code - either upon the early `return` here or a few lines down. Besides introducing a memory leak, I don't see any benefit in switching to a device-managed allocation in this case.